src/NITTA/Model/Problems/Allocation.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{- |
+Module      : NITTA.Model.Problems.Allocation
+Description : PU allocation on the bus network
+Copyright   : (c) Aleksandr Penskoi, 2022
+License     : BSD3
+Maintainer  : aleksandr.penskoi@gmail.com
+Stability   : experimental
+-}
+module NITTA.Model.Problems.Allocation (
+    Allocation (..),
+    AllocationProblem (..),
+) where
+
+import Data.String.ToString (ToString (..))
+import GHC.Generics (Generic)
+
+data Allocation tag = Allocation
+    { bnTag :: tag
+    -- ^Tag of the BusNetwork where PU will be allocated
+    , puTag :: tag
+    -- ^Tag of the prototype that will be used for allocation
```

Comment:
Can you rename it, because right now it's a little bit confusing. bn and pu looks like a prefix for variable name. networkTag and processUnitTag will be better.